LibreSSL doesn't have BIO_ADDR support, so this exception cannot be hit.

Index: src/crypto/OpenSSLHelpers.h
--- src/crypto/OpenSSLHelpers.h.orig
+++ src/crypto/OpenSSLHelpers.h
@@ -118,10 +118,12 @@ class OpenSSLException : public Exception
             while((error = ERR_get_error()) != 0)
             {
                 Exception e(ERR_lib_error_string(error), 0, ERR_error_string(error, nullptr));
+#ifndef LIBRESSL_VERSION_NUMBER
                 if(ERR_GET_LIB(error) == ERR_R_BIO_LIB &&
                     ERR_GET_FUNC(error) == BIO_F_BIO_LOOKUP_EX &&
                     ERR_GET_REASON(error) == ERR_R_SYS_LIB)
                     e.setCode(ExceptionCode::HostNotFound);
+#endif
                 addCause(e);
             }
         }
